crdt/sequence.py

```diff
@@ -309,8 +341,16 @@ def do(self, objects):
                 raise IndexError("Could not find object to remove")
             objects[index].tombstone = True
 
+    def __eq__(self, other):
+        if not isinstance(other, Operation):
+            return False
+        return self.owner == other.owner
+
+    def __hash__(self):
+        return hash(self.owner.get_hash())
+
```

Comment:
This is required to prevent putting duplicate operations in the log. The custom `__eq__` and `__hash__` functions are used by `set()` to determine distinctness when doing insertions. Without this, when notebooks get pickled and unpickled for transmission the CRDT can no longer tell if two operations are the same which breaks the logic.